test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs

```diff
@@ -356,18 +357,144 @@ public void InferBindingSourceForParameter_ReturnsBodyForSimpleTypes()
+            var action = GetActionModel(typeof(ParameterBindingController), actionName);
+            var options = new ApiBehaviorOptions
+            {
+                SuppressAddingConsumesConstraintForFormParameters = true,
+                InvalidModelStateResponseFactory = _ => null,
+            };
+            var provider = GetProvider(options);
+
+            // Act
+            provider.AddMultipartFormDataConsumesAttribute(action);
+
+            // Assert
+            var attribute = Assert.Single(action.Filters);
+            var consumesAttribute = Assert.IsType<ConsumesAttribute>(attribute);
+            Assert.Equal("application/json", Assert.Single(consumesAttribute.ContentTypes));
+        }
+
+        [Fact]
+        public void AddMultipartFormDataConsumesAttribute_AddsConsumesAttribute_WhenActionHasFromFormParameter()
+        {
+            // Arrange
+            var actionName = nameof(ParameterBindingController.FromFormParameter);
+            var action = GetActionModel(typeof(ParameterBindingController), actionName);
+            var provider = GetProvider();
+
+            // Act
+            provider.AddMultipartFormDataConsumesAttribute(action);
+
+            // Assert
+            var attribute = Assert.Single(action.Filters);
+            var consumesAttribute = Assert.IsType<ConsumesAttribute>(attribute);
+            Assert.Equal("multipart/form-data", Assert.Single(consumesAttribute.ContentTypes));
+        }
+
+        [Fact]
+        public void AddMultipartFormDataConsumesAttribute_AddsConsumesAttribute_WhenActionHasFromFormFileParameter()
+        {
+            // Arrange
+            var actionName = nameof(ParameterBindingController.FormFileParameter);
+            var action = GetActionModel(typeof(ParameterBindingController), actionName);
+            action.Parameters[0].BindingInfo = new BindingInfo
+            {
+                BindingSource = BindingSource.FormFile,
+            };
+            var provider = GetProvider();
+
+            // Act
+            provider.AddMultipartFormDataConsumesAttribute(action);
+
+            // Assert
+            var attribute = Assert.Single(action.Filters);
+            var consumesAttribute = Assert.IsType<ConsumesAttribute>(attribute);
+            Assert.Equal("multipart/form-data", Assert.Single(consumesAttribute.ContentTypes));
+        }
+
         private static ApiBehaviorApplicationModelProvider GetProvider(
-            IOptions<ApiBehaviorOptions> options = null,
+            ApiBehaviorOptions options = null,
             IModelMetadataProvider modelMetadataProvider = null)
         {
-            options = options ?? new TestOptionsManager<ApiBehaviorOptions>(new ApiBehaviorOptions
+            options = options ?? new ApiBehaviorOptions
             {
                 InvalidModelStateResponseFactory = _ => null,
-            });
+            };
+            var optionsProvider = new TestOptionsManager<ApiBehaviorOptions>(options);
```

Comment:
use Options.Create - we don't need our thing anymore

Reply:
https://github.com/aspnet/Mvc/pull/7005